### PR TITLE
Feature/index resource

### PR DESCRIPTION
**Pull request by gunther-bachmann**

### CODE REVIEW

File: org.testeditor.web.backend.xtext/src/main/java/org/testeditor/web/backend/xtext/StepTreeGenerator.xtend [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+	@Inject TclQualifiedNameProvider tclQualifiedNameProvider
+	@Inject ModelUtil modelUtil
+
+	def SerializableStepTreeNode generateStepTree() {
+		val allExportedObjects = resourceDescriptionsProvider.getResourceDescriptions(resourceDescriptionsProvider.indexResourceSet).exportedObjects.map[EObjectOrProxy]
+		val namespaceMacroCollectionsMap = allExportedObjects.namespaceMacroCollectionMap
+		val namespaceComponentsMap = allExportedObjects.namespaceComponentsMap
+		val namespaceObjectsMap = groupMerge(namespaceComponentsMap, namespaceMacroCollectionsMap)
+		val resultTree = new SerializableStepTreeNode => [
+			displayName = 'root'
```
Comment (jjelschen) [inline org.testeditor.web.backend.xtext/src/main/java/org/testeditor/web/backend/xtext/StepTreeGenerator.xtend]: How about 'Test Steps', 'Available Test Steps', or something like that, instead? That would actually make sense to display at the frontend, and the root node would not have to be treated specially. (Which it currently isn't, so 'root' actually appears at the top of the tree…)

File: org.testeditor.web.backend.xtext/src/main/java/org/testeditor/web/backend/xtext/StepTreeGenerator.xtend [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+		}
+	}
+
+	private def Map<String, List<NamedElement>> getNamespaceToNamedElementsMap(Iterable<NamedElement> namedElements, Function1<NamedElement, String> generateKey) {
+		val resolvedNamedElements = namedElements.map[EcoreUtil2.resolve(it, resourceDescriptionsProvider.indexResourceSet)] //
+		.filter(NamedElement)
+		val namespaceMacroCollectionsMap = resolvedNamedElements.groupBy [ namedElement |
+			val key = generateKey.apply(namedElement)
+			if (key.nullOrEmpty) {
+				return "./."
```
Comment (jjelschen) [inline org.testeditor.web.backend.xtext/src/main/java/org/testeditor/web/backend/xtext/StepTreeGenerator.xtend]: I don't really get this literal, where is this coming from / what's the meaning?

**Review by jjelschen — APPROVED** [review]
one change suggestion, one question

File: org.testeditor.web.backend.xtext/src/main/java/org/testeditor/web/backend/xtext/StepTreeGenerator.xtend (hunk 2 above)
Comment (gunther-bachmann) [reply org.testeditor.web.backend.xtext/src/main/java/org/testeditor/web/backend/xtext/StepTreeGenerator.xtend]: it's the default namespace. couldn't think of something better.

**Pull request merged**